node/core/dispute-coordinator/src/import.rs

```diff
@@ -477,6 +477,35 @@ impl ImportResult {
 		self.is_freshly_concluded_against() || self.is_freshly_concluded_for()
 	}
 
+	// Returns the number of invalid own votes from the state provided. This function is a helper for
+	// `has_fresh_byzantine_threshold_against`
+	fn get_own_invalid_votes<T>(state: &CandidateVoteState<T>) -> usize {
+		state
+			.own_votes()
+			.map(|votes| {
+				votes.iter().fold(0, |acc, (_, (vote, _))| match vote {
+					DisputeStatement::Invalid(_) => acc + 1,
+					_ => acc,
+				})
+			})
+			.unwrap_or(0)
+	}
+
+	/// Whether or not the invalid vote count for the dispute went beyond the byzantine threshold
+	/// after the last import
+	pub fn has_fresh_byzantine_threshold_against(&self, byzantine_threshold: usize) -> bool {
+		// Total number of imported invalid votes + our own invalid votes
+		let imported_invalid_votes =
+			self.imported_invalid_votes() as usize + Self::get_own_invalid_votes(self.new_state());
+		// The number of invalid votes + our own invalid votes before the import
+		let old_invalid_votes = (self.imported_invalid_votes() as usize)
+			.saturating_sub(self.new_invalid_voters().len()) +
+			Self::get_own_invalid_votes(self.old_state());
```

Comment:
this does not make any sense to me. imported_invalid_votes is always equal to new_invalid_voters.len, hence this always results in just our own invalid votes.

Reply:
Then I got this wrong. My idea here was from all invalid votes to subtract the invalid votes from the last import, so that I get `old_invalid_votes`. Maybe the correct way to do this is with `old_state()`?

But then why `old_state()` is:
https://github.com/paritytech/polkadot/blob/344bbcb5f5d300316c767fac0d24912c56027f24/node/core/dispute-coordinator/src/import.rs#L426

While `new_state()` is:
https://github.com/paritytech/polkadot/blob/344bbcb5f5d300316c767fac0d24912c56027f24/node/core/dispute-coordinator/src/import.rs#L431

Reply:
I think I got it - in 'old_state' we only have `DisputeStatus` - I can't query votes like I want above, right?

Reply:
yep, that sounds right.